test/algorithms/classifiers/test_vqc.py

```diff
@@ -12,390 +12,251 @@
-            num_qubits=num_qubits,
-            feature_map=feature_map,
-            ansatz=ansatz,
-            quantum_instance=self.qasm_quantum_instance,
-        )
+    bfgs = L_BFGS_B(maxiter=5)
+    cobyla = COBYLA(maxiter=25)
+    optimizers = [bfgs, cobyla, None]
 
-        # construct data
-        num_samples = 5
-        # pylint: disable=invalid-name
-        X = algorithm_globals.random.random((num_samples, num_inputs))
-        y = 1.0 * (np.sum(X, axis=1) <= 1)
-        while len(np.unique(y)) == 1:
-            X = algorithm_globals.random.random((num_samples, num_inputs))
-            y = 1.0 * (np.sum(X, axis=1) <= 1)
-        y = np.array([y, 1 - y]).transpose()  # VQC requires one-hot encoded input
+    losses = ["squared_error", "absolute_error", "cross_entropy"]
 
-        # fit to data
-        classifier.fit(X, y)
+    binary_x, binary_y = _generate_classification_data(6, 2, num_features)
 
-        # score
-        score = classifier.score(X, y)
-        self.assertGreater(score, 0.5)
+    binary_datasets = [
+        (2, binary_x, _one_hot_encode(binary_y)),
+    ]
 
-    @data(
-        # optimizer, quantum instance
-        ("cobyla", "statevector"),
-        ("cobyla", "qasm"),
-        ("bfgs", "statevector"),
-        ("bfgs", "qasm"),
-        (None, "statevector"),
-        (None, "qasm"),
+    num_classes = 3
+    multiclass_x, multiclass_y = _generate_classification_data(4, num_classes, num_features)
+    multiclass_datasets = [
+        (num_classes, multiclass_x, _one_hot_encode(multiclass_y)),
+    ]
+    datasets = binary_datasets + multiclass_datasets
+
+    def setUp(self):
+        super().setUp()
+        self.num_classes_by_batch = []
+
+    @idata(
+        itertools.product(
+            quantum_instances, num_qubits_list, feature_maps, ansatzes, optimizers, datasets
+        )
     )
-    def test_multiclass(self, config):
-        """Test multiclass VQC."""
-        opt, q_i = config
-
-        if q_i == "statevector":
-            quantum_instance = self.sv_quantum_instance
-        elif q_i == "qasm":
-            quantum_instance = self.qasm_quantum_instance
-        else:
-            quantum_instance = None
-
-        if opt == "bfgs":
-            optimizer = L_BFGS_B(maxiter=5)
-        elif opt == "cobyla":
-            optimizer = COBYLA(maxiter=25)
-        else:
-            optimizer = None
-
-        num_inputs = 2
-        feature_map = ZZFeatureMap(num_inputs)
-        ansatz = RealAmplitudes(num_inputs, reps=1)
-        # fix the initial point
-        initial_point = np.array([0.5] * ansatz.num_parameters)
+    @unpack
+    def test_VQC(self, quantum_instance, num_qubits, feature_map, ansatz, optimizer, dataset):
+        """
+        Test VQC with binary and multiclass data using a range of quantum
+        instances, numbers of qubits, feature maps, and optimizers.
+        """
+        if num_qubits is None and feature_map is None and ansatz is None:
+            return
```

Comment:
skipTest with a comment explaining that the arguments are None is better because then you know the full test didn't run, otherwise it will just show success in both cases.